app/src/main/java/com/sunasterisk/travelapp/ui/list/hotel/HotelListActivity.kt

```diff
@@ -0,0 +1,61 @@
+package com.sunasterisk.travelapp.ui.list.hotel
+
+import android.view.MenuItem
+import com.sunasterisk.travelapp.R
+import com.sunasterisk.travelapp.ui.adapter.HotelAdapter
+import com.sunasterisk.travelapp.ui.base.BaseActivity
+import com.sunasterisk.travelapp.ui.custom.TabName
+import com.sunasterisk.travelapp.utils.location
+import com.sunasterisk.travelapp.utils.screenHeight
+import kotlinx.android.synthetic.main.activity_hotel_list.*
+import kotlinx.android.synthetic.main.include_hotel_list_front.*
+
+class HotelListActivity : BaseActivity() {
+
+    override val layoutResource = R.layout.activity_hotel_list
```

Comment:
2 dòng sau có gì khác nhau nhỉ? 
```suggestion
    override val layoutResource = R.layout.activity_hotel_list
    override val layoutResource get() = R.layout.activity_hotel_list
```

Reply:
Ok ông, 
line 15: khai báo property với val thì Compiler sẽ tự động tạo ra default getter cho nó (ko có setter)
line 16: Tương tự line 15, custom getter với inferred type là @LayoutRes (Kotlin 1.1 trở lên)

Reply:
Ông thử xem bytecode nó xem sao nhé
Nếu bỏ get() => 
lớp kế thừa sẽ có dạng 
``` kotlin
class A {
   private int a = 123; 
   int getA(){ return this.a }
}
```
nếu thêm get() 
``` kotlin
class A {
    int getA(){ return 123; }
}
```
và annotation @get:LayoutRes chỉ sẽ check khi ta thêm get() 

Trong trường hợp này mình chỉ cần getter của nó mà thôi + check Res => thêm get()

Reply:
Vì bản chất các hai hàm get này đều trả ra các data object (Wrap từ primitivetype Int) nên hashCode() của nó đều bằng nhau mà

Reply:
ý t là nó sẽ sinh thêm những đoạn code không cần thiết + không check tại thời điểm buildtime khi dùng @ANNOTATION @get:LayoutRes

Reply:
ok ông